src/resource.ts

```diff
@@ -0,0 +1,312 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import LinkHeader from "http-link-header";
+import {
+  UrlString,
+  WithResourceInfo,
+  unstable_WithAcl,
+  unstable_WithAccessibleAcl,
+  unstable_AclDataset,
+  unstable_hasAccessibleAcl,
+  unstable_Access,
+} from "./interfaces";
+import { saveLitDatasetAt } from "./litDataset";
+import { fetch } from "./fetcher";
+import { internal_fetchResourceAcl, internal_fetchFallbackAcl } from "./acl";
+
+/** @internal */
+export const internal_defaultFetchOptions = {
+  fetch: fetch,
+};
+
+/**
+ * Retrieve the information about a resource (e.g. access permissions) without
+ * fetching the resource itself.
+ *
+ * @param url URL to fetch Resource metadata from.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters).
+ * @returns Promise resolving to the metadata describing the given Resource, or rejecting if fetching it failed.
+ */
+export async function internal_fetchResourceInfo(
```

Comment:
I just realised that we don't export this function publicly. Is it something that we want to expose along `unstable_fetchResourceInfo` ?

Reply:
I suppose we do, but I think we decided against it for now because it wasn't actually used and we weren't sure about the return type (being `{ resourceInfo: { /* the actual resourceInfo */ } }`).

Reply:
Oh right, I remember this discussion now. You're right, as long as nobody needs it, we should keep it internal (non-breaking change and all that)